### ISSUE TITLE
UnitTestService rebuilds tests multiple times on solution open

### ISSUE THREAD

**Issue by Therzok** [mono/monodevelop#6243]
Just freshly opened solution had tests built 4 times.

```
0.31%   <RebuildTests>b__0  •  5,013 ms  •  4 calls  •  MonoDevelop.UnitTesting.UnitTestService+<>c__DisplayClass30_0.<RebuildTests>b__0
  0.31%   BuildTest  •  5,013 ms  •  304 calls  •  MonoDevelop.UnitTesting.UnitTestService.BuildTest(WorkspaceObject)
    0.31%   CreateUnitTest  •  5,012 ms  •  304 calls  •  MonoDevelop.UnitTesting.SystemTestProvider.CreateUnitTest(WorkspaceObject)
      0.31%   get_HasTests  •  5,003 ms  •  144 calls  •  MonoDevelop.UnitTesting.UnitTestGroup.get_HasTests
        0.31%   get_Tests  •  5,002 ms  •  216 calls  •  MonoDevelop.UnitTesting.UnitTestGroup.get_Tests
          0.31%   OnCreateTests  •  5,002 ms  •  70 calls  •  MonoDevelop.UnitTesting.SolutionFolderTestGroup.OnCreateTests
             0.00%   6 functions hidden  •  1 ms total  •  954 calls total
          ►0.00%   UnitTestCollection..ctor  •  0 ms  •  70 calls  •  MonoDevelop.UnitTesting.UnitTestCollection..ctor(UnitTestGroup)
         0.00%   5 functions hidden  •  0 ms total  •  286 calls total
       0.00%   3 functions hidden  •  48 ms total  •  116 calls total
    0.30%   CreateUnitTest  •  4,857 ms  •  232 calls  •  MonoDevelop.UnitTesting.VsTest.VsTestTestProvider.CreateUnitTest(WorkspaceObject)
      0.30%   GetTestAdapters  •  4,856 ms  •  186 calls  •  MonoDevelop.UnitTesting.VsTest.VsTestAdapter.GetTestAdapters(Project)
        ►0.24%   GetInstalledPackages  •  3,993 ms  •  186 calls  •  MonoDevelop.PackageManagement.PackageManagementProjectOperations.GetInstalledPackages(Project)
        ►0.05%   GetFiles  •  852 ms  •  394 calls  •  System.IO.Directory.GetFiles(String, String, SearchOption)
         0.00%   19 functions hidden  •  10 ms total  •  2463 calls total
       0.00%   3 functions hidden  •  0 ms total  •  188 calls total
    ►0.01%   CreateUnitTest  •  101 ms  •  278 calls  •  MonoDevelop.UnitTesting.NUnit.SystemTestProvider.CreateUnitTest(WorkspaceObject)
     0.00%   3 functions hidden  •  0 ms total  •  2164 calls total
   0.00%   2 functions hidden  •  0 ms total  •  4 calls total
```

> VS bug [#700296](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/700296)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-17] Comment by **xamarinc**



Fixed in version 7.8.0.1108 (master)



Pull Request #6405 merged by: Mike Krüger  
 Author: mono  
 Commit: 54f76816e87b3cc3ada36205c803178c8917dfaf (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-23] Comment by **xamarinc**



Fixed in version 8.0.0.604 (release-8.0)



Pull Request #6405 merged by: Mike Krüger  
 Author: mono  
 Commit: 54f76816e87b3cc3ada36205c803178c8917dfaf (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-26] Comment by **xamarinc**



Fixed in version 7.8.0.1193 (release-7.8)



Pull Request #6405 merged by: Mike Krüger  
 Author: mono  
 Commit: 54f76816e87b3cc3ada36205c803178c8917dfaf (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-30] Comment by **xamarinc**



Fixed in version 8.0.0.1738 (release-8.0)



Pull Request #6405 merged by: Mike Krüger  
 Author: mono  
 Commit: 54f76816e87b3cc3ada36205c803178c8917dfaf (xamarin/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-48] Comment by **xamarinc**



Fixed in version 8.6.0.802 (master)



Pull Request #6405 merged by: Mike Kr??ger  
 Author: mono  
 Commit: 54f76816e87b3cc3ada36205c803178c8917dfaf (xamarin/monodevelop)
